nifi-nar-bundles/nifi-enrich-bundle/nifi-enrich-processors/src/main/java/org/apache/nifi/processors/enrich/QueryWhois.java

```diff
@@ -0,0 +1,334 @@
+                        flowFile = session.putAllAttributes(flowFile, parsedResults);
+                        flowFilesMatched.add(flowFile);
+
+                        // Finished processing single result
+
+                } else {
+                    // Otherwise call the multiline parser and get the row map;
+                    final Map<String, Map<String, String>> rowMap = parseBatchResponse(result, queryParser, queryRegex, keyLookup, "whois").rowMap();
+
+                    // Identify the flowfile Lookupvalue and search against the rowMap
+                    String ffLookupValue = context.getProperty(QUERY_INPUT).evaluateAttributeExpressions(flowFile).getValue();
+
+                    if (rowMap.containsKey(ffLookupValue)) {
+                        // flowfile Lookup Value is contained within the results, get the properties and add to matched list
+                        flowFile = session.putAllAttributes(flowFile, rowMap.get(ffLookupValue));
+                        flowFilesMatched.add(flowFile);
+                    } else {
+                        // otherwise add to Not Matched
+                        flowFilesNotMatched.add(flowFile);
+                    }
+                }
+            }
+        }
+
+        // Finally prepare to send the data down the pipeline
+        // Because batches may include matches and non-matches, test both and send
+        // each to its relationship
+        if (flowFilesMatched.size() > 0) {
+            // Sending the resulting flowfile (with attributes) to REL_FOUND
+            session.transfer(flowFilesMatched, REL_FOUND);
+        }
+        if (flowFilesNotMatched.size() > 0) {
+            // Sending whatetver didn't match to REL_NOT_FOUND
+            session.transfer(flowFilesNotMatched, REL_NOT_FOUND);
+        }
+    }
+
+    /**
+     * This method performs a simple Whois lookup
+     * @param whoisServer Server to be queried;
+    *  @param whoisPort TCP port to be useed to connect to server
+     * @param whoisTimeout How long to wait for a response (in ms);
+     * @param query The query to be made;
+     */
+    protected String doLookup(String whoisServer, int whoisPort, int whoisTimeout, String query) {
+        // This is a simple WHOIS lookup attempt
+
+        String result = null;
+
+        whoisClient = createClient();
+
+        try {
+            // Uses pre-existing context to resolve
+            if (!whoisClient.isConnected()) {
+                whoisClient.connect(whoisServer, whoisPort);
+                whoisClient.setSoTimeout(whoisTimeout);
+                result = whoisClient.query(query);
+                // clean up...
+                if (whoisClient.isConnected()) whoisClient.disconnect();
+            }
+        } catch ( IOException e) {
+            getLogger().error("Query failed due to {}", new Object[]{e});
+            throw new ProcessException("Error performing Whois Lookup {}", e.getCause());
```

Comment:
When catching an exception and re-throwing a new one, I think this is best to always reference the original exception as-is:
````java
throw new ProcessException("Error performing Whois Lookup", e);
````